rust/codec-json/src/lib.rs

```diff
@@ -176,9 +176,10 @@ pub fn to_path(node: &Node, path: &Path, options: Option<EncodeOptions>) -> Resu
         let filename = path
             .file_name()
             .and_then(|name| name.to_str())
-            .and_then(|name| name.split('.').next())
+            .and_then(|name| name.rsplit_once('.'))
+            .and_then(|name| Some(name.0))
             .unwrap_or("document");
-        let filename = [filename, ".json"].concat();
+        // let filename = [filename, ".json"].concat();
```

Comment:
That is what I thought but when I ran it with that line I got a .json.json file out and without it got the .json file as it should be, I am not sure why.

Reply:
This is because when it was `name.split('.').next()` it would get just the first element so in `"name.json.zip"` i would get just `"name"` so `.json` needed to be added to the end. 
![Screenshot 2024-11-27 at 14 47 00](https://github.com/user-attachments/assets/bb4b14a7-eb5e-45dc-b435-3ab153972e9a)

While `name.rsplit_once('.').0` gets every thing before the split on the last `.` so in `"name.json.zip"` it would get `"name.json"` so `.json` doen't need to be added to the end
![Screenshot 2024-11-27 at 14 50 51](https://github.com/user-attachments/assets/028d9090-3b5d-4c39-905c-0c4e4551a9df)

Reply:
Right, make sense! Can you remove that commented out line then please.